dataflow_test.py

```diff
@@ -490,9 +490,13 @@ def test_slice_raises_ValueError(args):
         df.slice(*args)
 
 
-@mark.xfail
 def test_pipes_must_end_in_a_sink():
-    raise NotImplementedError
+    the_source    = range(10)
+    sinkless_pipe = df.map(abs)
+
+    with raises(AttributeError):
```

Comment:
We should make some effort to make the message in this error be as helpful as possible. Or even raise our own specific exception type.

Reply:
Perhaps this should be listed in an issue which also mentions the points that @mmkekic made (in the thread in IC) about error clarity.